src/__experimental__/components/date/date-picker.component.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import 'react-day-picker/lib/style.css';
+import LocaleUtils from 'react-day-picker/moment';
+import Browser from '../../../utils/helpers/browser/browser';
+import DateHelper from '../../../utils/helpers/date/date';
+import Portal from '../../../components/portal/portal';
+import Navbar from './navbar';
+import StyledDayPicker from './day-picker.style';
+
+const DatePicker = (props) => {
+  const window = Browser.getWindow();
+  const [containerPosition, updateDatePickerPosition] = useState(getContainerPosition(window, props.inputElement));
```

Comment:
Just curious about what actually happens on each re-render if you use a variable as the initial value in `useState`? Does it ignore the `getContainerPosition` function on each re-render?

Reply:
the `useState` method sets the initial value of the state, so the `getContainerPosition` method is called only once. `getContainerPosition` returns a new object every time it's called, but even if we'll provide to the `setState` method an object defined as a variable it will deep copy that object, so further modifications should not change the original.

Reply:
Looking into it it looks as though `getContainerPosition` is called on every new render (eg. when a prop or state changes). I think you need to wrap it in a callback to trigger only once

```js
const [containerPosition, updateDatePickerPosition] = useState(() => getContainerPosition(window, props.inputElement))
```